main.go

```diff
@@ -87,7 +89,7 @@ func main() {
 	l, err := zap.NewDevelopment()
 	if err != nil {
 		fmt.Printf("failed to set up logger\n")
-		os.Exit(2)
+		os.Exit(1)
```

Comment:
This jumps out at me as not related to the rest of the patch. Is there a reason for this change?

Reply:
Try to have different exit code on different failures (1, 2 and 3) upon boot sequence. Maybe it deserves a dedicate PR for clarity.

Reply:
IMO a separate patch/commit in this PR would be sufficient to make me happy. Along with that explanation in the commit msg. :-)

Reply:
Fair enough. Will fix that